lib/resty/jwt.lua

```diff
@@ -466,9 +496,11 @@ function _M.sign(self, secret_key, jwt_obj)
   local alg = jwt_obj[str_const.header][str_const.alg]
   local signature = ""
   if alg == str_const.HS256 then
-    signature = hmac:new(secret_key, hmac.ALGOS.SHA256):final(message)
+    local secret_str = get_secret_str(secret_key, jwt_obj)
```

Comment:
Just as a note - unit test number 3 tests the case where kid is missing, and test number 4 covers the case where the kid function returns an error itself.

As I'm thinking about it, I think that `sign` actually *should* raise an error...because if you call `sign` directly with the headers you want, you aren't decoding it.  And `sign` is a actually wrapped in a `pcall` where it is used within `verify_jwt_obj`.